src/Sentry/Internal/FileSystem.cs

```diff
@@ -16,38 +19,112 @@ public IEnumerable<string> EnumerateFiles(string path, string searchPattern) =>
     public IEnumerable<string> EnumerateFiles(string path, string searchPattern, SearchOption searchOption) =>
         Directory.EnumerateFiles(path, searchPattern, searchOption);
 
-    public void CreateDirectory(string path) => Directory.CreateDirectory(path);
+    public DirectoryInfo? CreateDirectory(string path)
+    {
+        if (!_options?.DisableFileWrite is false)
+        {
+            _options?.LogDebug("Skipping creating directory. Writing to file system has been explicitly disabled.");
+            return null;
+        }
+
+        return Directory.CreateDirectory(path);
+    }
+
+    public bool? DeleteDirectory(string path, bool recursive = false)
+    {
+        if (!_options?.DisableFileWrite is false)
+        {
+            _options?.LogDebug("Skipping deleting directory. Writing to file system has been explicitly disabled.");
+            return false;
+        }
 
-    public void DeleteDirectory(string path, bool recursive = false) => Directory.Delete(path, recursive);
+        Directory.Delete(path, recursive);
+        return !Directory.Exists(path);
+    }
 
     public bool DirectoryExists(string path) => Directory.Exists(path);
 
     public bool FileExists(string path) => File.Exists(path);
 
-    public void MoveFile(string sourceFileName, string destFileName, bool overwrite = false)
+    public bool? MoveFile(string sourceFileName, string destFileName, bool overwrite = false)
     {
+        if (!_options?.DisableFileWrite is false)
+        {
+            _options?.LogDebug("Skipping moving file. Writing to file system has been explicitly disabled.");
+            return null;
```

Comment:
This is inconsistent with the behaviour in `DeleteDirectory`... In that method, if `DisableFileWrite` is enabled then a value of `false` is returned. Whereas, this method returns `null` when `DisableFileWrite` is enabled. 

Whatever we're deciding null and false represent, I think we should be consistent. 

Given that, possibly worth changing the return value to be an enum (Succeeded, Failed, Disabled) so that there is no ambiguity over what happened or why.